images/kcp-binder/Dockerfile

```diff
@@ -0,0 +1,42 @@
+# Build the binary
+FROM golang:1.17 AS builder
+
+WORKDIR /workspace
+USER 0
+RUN apt-get update && apt-get install -y jq && mkdir bin
+RUN git clone https://github.com/kcp-dev/kcp.git && cd kcp && \
+    BRANCH=release-0.4 && git checkout $BRANCH && \
+    CGO_ENABLED=0 make
+
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.6
+LABEL build-date= \
+      com.redhat.build-host= \
+      description="This image provides binaries and a script to easily register clusters to kcp." \
+      distribution-scope="public" \
+      io.k8s.description="This image provides binaries and a script to easily register clusters to kcp." \
+      io.k8s.display-name="kcp register" \
+      maintainer="Pipelines Service" \
+      name="kcp-binder" \
+      release="0.1" \
+      summary="Provides the latest release of kcp-binder image." \
+      url="https://github.com/openshift-pipelines/pipelines-service/tree/main/images/kcp-binder" \
+      vcs-ref=  \
+      vcs-type="git" \
+      vendor="Pipelines Service" \
+      version="0.1"
+WORKDIR /
+RUN mkdir /workspace && chmod 777 /workspace && chown 65532:65532 /workspace
+COPY --from=builder workspace/kcp/bin/kubectl-kcp /usr/local/bin/kubectl-kcp
+RUN chmod +rx /usr/local/bin/kubectl-kcp
```

Comment:
+rx does that: add read and execute, not write. Also I don't think it is really a security issue. If the binary was modified as part of the PipelineRun it would be reverted to the previous version by the next run.

Reply:
`+rx` does not set the `w` bit, so if the script is writable `chmod +rx` does not make it unwriteable.  I'd rather we set all bits explicitly.

Reply:
The default umask for a standard user is 0002. Anyways I have done the change